src/dune/lib.ml

```diff
@@ -1542,7 +1564,9 @@ module Compile = struct
       t.requires
     in
     let lib_deps_info =
-      let pps = Lib_info.pps t.info in
+      let pps =
+        Result.ok_exn t.pps |> List.map ~f:(fun t -> (Loc.none, t.name))
```

Comment:
`--debug-backtrace` pointed to this call

Reply:
To give a bit of context, `lib_deps_info` is used to compute `dune external-lib-deps`. So essentially we should include:

- the raw list of pps as before
- for instrumentation backends that are enabled, we can report the ppx part of the backend, or at least the backend name if the library is not available

Reply:
Thanks for the detective work. I will look into it.

Reply:
I added 308c17533822cf96b86deccfb14011292bf6fb40 with this. The only issue is that I needed to put back a line of the form `let db = Option.value_exn db` which I had been able to remove in a previous simplification suggested by @rgrinberg. I still don't understand very well if this is safe to do, so it would be great to have a second opinion on that.